src/main/java/com/example/demo/service/ShedulerService.java

```diff
@@ -0,0 +1,67 @@
+package com.example.demo.service;
+
+import com.example.demo.model.Message;
+import com.example.demo.model.Status;
+import com.example.demo.repo.JsonRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component
+@EnableScheduling
+public class ShedulerService {
+
+    private JsonRepo jsonRepo;
+    private Set<Message> messageListDoNotSend;
+    private List<Message> messageLists;
+    private Message message;
+    private MailSender messageService;
+
+    @Autowired
+    public ShedulerService(JsonRepo jsonRepo, MailSender messageService) {
+        this.messageService = messageService;
+        this.jsonRepo = jsonRepo;
+        this.messageListDoNotSend = new LinkedHashSet<>();
+        this.messageLists = jsonRepo.loadFromJsonToList();
+        this.message = new Message();
+    }
+
+    @Scheduled(fixedDelay = 1000)
+    private void dataCheck() {
+        if (messageListDoNotSend.size() != 0) {
+            sendMessageInFuture();
+        }
+        if (messageLists.size() != 0) {
+            checkStatus();
+        }
+    }
+
+    private void sendMessageInFuture() {
+        Date currentDate = new Date();
+        Iterator iter = messageListDoNotSend.iterator();
```

Comment:
All modifications to messages should be done via repository. It's out of scope of this class

Reply:
Sorry i did not understand
Later I invoke a changeStatus method that makes changes to the repository
Please, specify the question

Reply:
1) I don't see a reason to have an extra array/iterable in this service. You should call repository when you need the data;
2) repository should be capable of updating single record. you don't need to rewrite whole list;

Reply:
Done